pdr-api/migrations/2023-11-24_updateEstablishedStatus.js

```diff
@@ -0,0 +1,100 @@
+const AWS = require('aws-sdk');
+const { updateConsoleProgress, finishConsoleUpdates, errorConsoleUpdates } = require('../tools/progressIndicator');
+const TABLE_NAME = process.env.TABLE_NAME || 'NameRegister';
+
+/*
+  This migration pulls all the protectedArea items in the database and performs the following updates on each item:
+
+    - status: current -> status: established
+    - status: old -> status: historical
+    - new attribute: type: protectedArea
+
+  WARNING: at the time of running this script (2023-11-24), protected areas are the only item type in the database. It likely will not be appropriate to run this migration again in the future. This migration leverages DynamoDB scan.
+*/
+
+// The number of items a transaction can perform at once
+const BATCH_ITEM_LIMIT = 25;
+
+const options = {
+  region: 'ca-central-1'
+};
+
+if (process.env.IS_OFFLINE === 'true') {
+  options.endpoint = process.env.DYNAMODB_ENDPOINT_URL || 'http://localhost:8000';
+}
+
+const dynamodb = new AWS.DynamoDB(options);
+
+async function run() {
+  try {
+    // Get all the items in the database
+    const scan = {
+      TableName: TABLE_NAME
+    }
+    const items = await dynamodb.scan(scan).promise();
+
+    // Iterate through the items and update the necessary fields
+    let updates = [];
+    for (let item of items.Items) {
+      // Future proofing in case this is run again
+      // If the item has a type, this migration has already been run and should be aborted.
+      if (item?.type?.S) {
+        if (item?.type?.S !== 'protectedArea' && item?.type?.S !== 'protected-area'){
+          throw 'Encountered item with attribute `type` != `protectedArea`. This suggests this migration has already been run. Aborting migration.'
```

Comment:
Maybe log here and `continue;` as it may only be one of several records that have this problem.

Reply:
fixed in [cd3632c](https://github.com/bcgov/parks-data-register/pull/234/commits/cd3632cd4ed82c9db0460e3a91f32ef545d9a67d)